google-cloud-storage/lib/google/cloud/storage/bucket.rb

```diff
@@ -895,6 +895,85 @@ def policy_only_locked_at
           uniform_bucket_level_access_locked_at
         end
 
+        ##
+        # Whether the bucket's file IAM configuration enforces Public Access Prevention. Currently, `unspecified` and
+        # `enforced` are supported. When set to `enforced`, Public Access Prevention is enforced in the bucket's IAM
+        # configuration. This value can be modified by calling {Bucket#public_access_prevention=}.
+        #
+        # @return [String, nil] Currently, `unspecified` and `enforced` are supported. Returns `nil` if the bucket has
+        #    no IAM configuration.
+        #
+        # @example
+        #   require "google/cloud/storage"
+        #
+        #   storage = Google::Cloud::Storage.new
+        #
+        #   bucket = storage.bucket "my-bucket"
+        #
+        #   bucket.public_access_prevention = "enforced"
```

Comment:
Would it be typical to provide constants for "enforced" and "unspecified" in Ruby, or is it normal to require the user to supply the right string? I know different languages have different guidelines on this...

Reply:
Typically as a matter of Ruby style, we do not use constants in these client libraries, but instead require the user to set the right **symbol**. (In this case the symbol would be: (`bucket.public_access_prevention = :enforced`). Symbols are supported by this method, but in a departure from style, at the request of @shaffeeullah, I changed the examples to show string arguments instead.

Reply:
Gotcha. In your original implementation, was it also possible to set "unspecified" via a symbol? If so it might be fine to stick with that if that's what's typical in Ruby. I'll confer with Sameena.

Reply:
As an analog to this, I looked at the code for Storage Classes in this library and I _think_ it's using symbols for what is essentially a string enum field. Do you think something like that would make sense here?